main.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import schemas
+from crud import (
+    get_all_authors,
+    create_author,
+    get_author_detail,
+    get_all_books,
+    create_book,
+)
+from database import SessionLocal
+
+
+app = FastAPI()
+
+API_PREFIX = "/api/v1/"
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get(API_PREFIX + "authors/", response_model=list[schemas.Author])
+def read_all_authors(
+        skip: int | None = None,
+        limit: int | None = None,
+        db: Session = Depends(get_db)
+) -> list[schemas.Author]:
+    return get_all_authors(db=db, skip=skip, limit=limit)
+
+
+@app.post(API_PREFIX + "authors/", response_model=schemas.Author)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+) -> schemas.Author:
+    return create_author(db=db, author=author)
+
+
+@app.get(API_PREFIX + "authors/{id}/", response_model=schemas.Author)
+def read_author(
+        id: int,
+        db: Session = Depends(get_db)
+) -> schemas.Author:
+    author = get_author_detail(db=db, id=id)
+
+    if author is None:
+        raise HTTPException(
+            status_code=404,
+            detail=f"Author with id {id} not found!"
+        )
```

Comment:
This is a good use of FastAPI's `HTTPException` to handle errors when an author is not found. However, to avoid repetition, you could create a helper function that takes an instance and its id as parameters, checks if the instance is None and if so, raises the `HTTPException`. This function could be used in other endpoints as well.